blockonomics-woocommerce.php

```diff
@@ -84,18 +84,21 @@ function bnomics_exclude_pages( $exclude ) {
     }
 
     function add_payment_page_shortcode() {
-        $show_order = isset($_GET["show_order"]) ? sanitize_text_field(wp_unslash($_GET['show_order'])) : "";
-        $crypto = isset($_GET["crypto"]) ? sanitize_key($_GET['crypto']) : "";
-        $select_crypto = isset($_GET["select_crypto"]) ? sanitize_text_field(wp_unslash($_GET['select_crypto'])) : "";
-        $blockonomics = new Blockonomics;
+        $currentFilter = current_filter();
+        if ($currentFilter == 'the_content') {
```

Comment:
The fix is working well at solving the issues with the All in One SEO plugin. Let's move this validation to the top of the shortcode function to improve readability
```
$currentFilter = current_filter();
if ($currentFilter != 'the_content') {
    return;
}
```

Reply:
Let's also add a code comment to explain why we are including this check.